dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/EventExporterTest.java

```diff
@@ -317,6 +322,58 @@ void testExportEventsWithLastUpdateDates() throws ForbiddenException, BadRequest
     assertContainsOnly(List.of("D9PbzJY8bJM"), events);
   }
 
+  @Test
+  void testExportEventsWithDatesIncludingTimeStamp()
+      throws ForbiddenException, BadRequestException {
+    Date date = new Date();
+    EventOperationParams params =
+        EventOperationParams.builder()
+            .events(Set.of("pTzf9KYMk72"))
+            .updatedAfter(
+                Date.from(
+                    date.toInstant()
+                        .minus(1, ChronoUnit.DAYS)
+                        .atZone(ZoneId.systemDefault())
+                        .toInstant()))
+            .updatedBefore(
+                Date.from(
+                    date.toInstant()
+                        .plus(1, ChronoUnit.DAYS)
+                        .atZone(ZoneId.systemDefault())
+                        .toInstant()))
+            .build();
+
+    Events events = eventService.getEvents(params);
+
+    assertNotEmpty(events.getEvents());
+
+    Event event = events.getEvents().get(0);
+
+    assertAll(
+        "All dates should include timestamp",
+        () ->
+            assertEquals(
+                "2019-01-25T12:10:38.100",
+                DateUtils.getIso8601NoTz(event.getExecutionDate()),
+                () ->
+                    String.format(
+                        "Expected %s to be in %s",
+                        event.getExecutionDate(), "2019-01-25T12:10:38.100")),
+        () ->
+            assertEquals(
+                "2019-01-28T12:32:38.100",
+                DateUtils.getIso8601NoTz(event.getDueDate()),
+                () ->
+                    String.format(
+                        "Expected %s to be in %s", event.getDueDate(), "2019-01-28T12:32:38.100")),
+        () ->
```

Comment:
Why did you not assert  `completedDate` in the same way you asserted the other two fields? So

```suggestion
        () ->
            assertEquals(
                "2019-01-25T12:10:38.100",
                DateUtils.getIso8601NoTz(event.getCompletedDate()),
                () ->
                    String.format(
                        "Expected %s to be in %s",
                        event.getCompletedDate(), "2019-01-28T22:17:43.000")),
```

And adding a time to the `dhis-2/dhis-test-integration/src/test/resources/tracker/event_and_enrollment.json` `"completedAt": "2019-01-28T22:17:43.000"` 😅

Reply:
> Why did you not assert `completedDate` in the same way you asserted the other two fields? So
> 
> And adding a time to the `dhis-2/dhis-test-integration/src/test/resources/tracker/event_and_enrollment.json` `"completedAt": "2019-01-28T22:17:43.000"` 😅

https://github.com/dhis2/dhis2-core/pull/14628#discussion_r1272153931

Reply:
checking if it has a timestamp and checking if it has exactly the same timestamp are two different things

Reply:
moreover, it sets `completedDate` when it processes this event so an exact comparison would be difficult in this case.

Reply:
What is setting the `completedDate`?

Please remove any timestamps that are automatically set from the JSON test fixture. Otherwise, its really hard to review the PR and understand the test in the future 😅

Reply:
the service layer is setting `completedDate` at the time of changing the event status to `COMPLETED`

Reply:
having a timestamp in completedDate is important and this is exactly what I am checking in the test using `hasTimeStamp`